### ISSUE TITLE
WeightWatcher produces an empty output DataFrame

### ISSUE THREAD

**Issue by madarax64** [CalculatedContent/WeightWatcher#322]
Hello,
I was trying to use WeightWatcher to analyze a saved Tensorflow model (saved in the .h5 format). However WeightWatcher appears to produce an empty DataFrame so I don't receive any metrics. Here is what I get:
```
Import error No module named 'torch', reetting to svd accurate methods
I0000 00:00:1749456129.391660  948280 gpu_device.cc:2019] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46863 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:1f:00.0, compute capability: 8.6
2025-06-09 12:02:11,594 - WARNING - Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.
2025-06-09 12:02:11,622 - WARNING - Error in loading the saved optimizer state. As a result, your model is starting with a freshly initialized optimizer.
2025-06-09 12:02:11,629 - INFO -  __init__: <Functional name=model, built=True>, None
2025-06-09 12:02:11,629 - INFO -

python      version 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 08:45:29)
[GCC 10.4.0]
numpy       version 1.26.4
tensorflow version 2.19.0

2025-06-09 12:02:11,629 - INFO - SET MODEL: None, None
2025-06-09 12:02:11,630 - INFO - params {'glorot_fix': False, 'normalize': False, 'conv2d_norm': True, 'randomize': True, 'savedir': 'ww-img', 'savefig': 'ww-img', 'rescale': True, 'plot': False, 'deltaEs': False, 'intra': False, 'channels': None, 'fft': False, 'conv2d_fft': False, 'ww2x': False, 'pool': True, 'vectors': True, 'smooth': None, 'stacked': False, 'svd_method': 'fast', 'fix_fingers': None, 'fit': 'power_law', 'MAX_FINGERS': 10, 'finger_thresh': 1.0, 'sparsify': True, 'detX': True, 'mp_fit': False, 'min_evals': 10, 'max_evals': 15000, 'max_N': 50000, 'tolerance': 1e-06, 'layer_ids_start': 0, 'add_biases': False, 'xmin_max': None, 'pl_package': 'ww', 'xmax': None, 'peft': False, 'inverse': False, 'glorot_fit': False, 'layers': []}
2025-06-09 12:02:11,630 - INFO - Saving all images to ww-img
2025-06-09 12:02:11,630 - INFO - SET MODEL: <Functional name=model, built=True>, None
2025-06-09 12:02:11,630 - INFO - params {'glorot_fix': False, 'normalize': False, 'conv2d_norm': True, 'randomize': True, 'savedir': 'ww-img', 'savefig': 'ww-img', 'rescale': True, 'plot': False, 'deltaEs': False, 'intra': False, 'channels': None, 'fft': False, 'conv2d_fft': False, 'ww2x': False, 'pool': True, 'vectors': True, 'smooth': None, 'stacked': False, 'svd_method': 'fast', 'fix_fingers': None, 'fit': 'power_law', 'MAX_FINGERS': 10, 'finger_thresh': 1.0, 'sparsify': True, 'detX': True, 'mp_fit': False, 'min_evals': 10, 'max_evals': 15000, 'max_N': 50000, 'tolerance': 1e-06, 'layer_ids_start': 0, 'add_biases': False, 'xmin_max': None, 'pl_package': 'ww', 'xmax': None, 'peft': False, 'inverse': False, 'glorot_fit': False, 'layers': []}
2025-06-09 12:02:11,630 - INFO - Saving all images to ww-img
2025-06-09 12:02:11,630 - INFO - Using <class 'weightwatcher.weightwatcher.WWLayerIterator'>
2025-06-09 12:02:11,630 - INFO - tensorflow version 2.19.0

2025-06-09 12:02:11,630 - INFO - framework from model = keras
2025-06-09 12:02:12,085 - INFO - keras layer: input_1 <InputLayer name=input_1, built=True>  type unknown not found
2025-06-09 12:02:12,097 - INFO - keras layer: batch_normalization_1 <BatchNormalization name=batch_normalization_1, built=True>  type unknown not found
2025-06-09 12:02:12,097 - INFO - keras layer: activation_1 <Activation name=activation_1, built=True>  type unknown not found
2025-06-09 12:02:12,101 - INFO - keras layer: batch_normalization_2 <BatchNormalization name=batch_normalization_2, built=True>  type unknown not found
2025-06-09 12:02:12,102 - INFO - keras layer: activation_2 <Activation name=activation_2, built=True>  type unknown not found
2025-06-09 12:02:12,107 - INFO - keras layer: batch_normalization_4 <BatchNormalization name=batch_normalization_4, built=True>  type unknown not found
2025-06-09 12:02:12,108 - INFO - keras layer: batch_normalization_3 <BatchNormalization name=batch_normalization_3, built=True>  type unknown not found
2025-06-09 12:02:12,108 - INFO - keras layer: add <Add name=add, built=True>  type unknown not found
2025-06-09 12:02:12,108 - INFO - keras layer: activation_3 <Activation name=activation_3, built=True>  type unknown not found
2025-06-09 12:02:12,112 - INFO - keras layer: batch_normalization_5 <BatchNormalization name=batch_normalization_5, built=True>  type unknown not found
2025-06-09 12:02:12,113 - INFO - keras layer: activation_4 <Activation name=activation_4, built=True>  type unknown not found
2025-06-09 12:02:12,115 - INFO - keras layer: batch_normalization_6 <BatchNormalization name=batch_normalization_6, built=True>  type unknown not found
2025-06-09 12:02:12,115 - INFO - keras layer: activation_5 <Activation name=activation_5, built=True>  type unknown not found
2025-06-09 12:02:12,119 - INFO - keras layer: batch_normalization_8 <BatchNormalization name=batch_normalization_8, built=True>  type unknown not found
2025-06-09 12:02:12,120 - INFO - keras layer: batch_normalization_7 <BatchNormalization name=batch_normalization_7, built=True>  type unknown not found
2025-06-09 12:02:12,120 - INFO - keras layer: add_1 <Add name=add_1, built=True>  type unknown not found
2025-06-09 12:02:12,120 - INFO - keras layer: activation_6 <Activation name=activation_6, built=True>  type unknown not found
2025-06-09 12:02:12,122 - INFO - keras layer: batch_normalization_9 <BatchNormalization name=batch_normalization_9, built=True>  type unknown not found
2025-06-09 12:02:12,122 - INFO - keras layer: activation_7 <Activation name=activation_7, built=True>  type unknown not found
2025-06-09 12:02:12,124 - INFO - keras layer: batch_normalization_10 <BatchNormalization name=batch_normalization_10, built=True>  type unknown not found
2025-06-09 12:02:12,124 - INFO - keras layer: activation_8 <Activation name=activation_8, built=True>  type unknown not found
2025-06-09 12:02:12,125 - INFO - keras layer: batch_normalization_12 <BatchNormalization name=batch_normalization_12, built=True>  type unknown not found
2025-06-09 12:02:12,126 - INFO - keras layer: batch_normalization_11 <BatchNormalization name=batch_normalization_11, built=True>  type unknown not found
2025-06-09 12:02:12,126 - INFO - keras layer: add_2 <Add name=add_2, built=True>  type unknown not found
2025-06-09 12:02:12,126 - INFO - keras layer: activation_9 <Activation name=activation_9, built=True>  type unknown not found
2025-06-09 12:02:12,126 - INFO - keras layer: global_average_pooling1d <GlobalAveragePooling1D name=global_average_pooling1d, built=True>  type unknown not found
2025-06-09 12:02:12,128 - WARNING - No layers found
Empty DataFrame
Columns: []
Index: []
```

As can be seen from above, the model is made up of standard Keras primitives (e.g. Conv1D (don't appear above), BatchNormalization, GlobalAveragePooling1D, etc) Can you kindly tell me how to resolve this? 

Thanks!

**Comment by charlesmartin14** [comment]
can you join the discord channel to discuss ?

**Comment by madarax64** [comment]
Sure, joined and posted on the help channel.
